src/utils/Cypher.ts

```diff
@@ -7,36 +8,48 @@ import LockingManager from '../locking/LockingManager';
 import SettingStorage from '../storage/mongodb/SettingStorage';
 import Utils from './Utils';
 import _ from 'lodash';
-import crypto from 'crypto';
 
 const IV_LENGTH = 16;
 const MODULE_NAME = 'Cypher';
 
 export default class Cypher {
-
   public static async encrypt(tenantID: string, data: string, useFormerKey = false, cryptoSetting?: CryptoSetting): Promise<string> {
     const iv = crypto.randomBytes(IV_LENGTH);
     if (!cryptoSetting) {
       cryptoSetting = (await Cypher.getCryptoSettings(tenantID)).crypto;
     }
     const algo = useFormerKey ? Utils.buildAlgorithm(cryptoSetting.formerKeyProperties) : Utils.buildAlgorithm(cryptoSetting.keyProperties);
     const key = useFormerKey ? Buffer.from(cryptoSetting.formerKey) : Buffer.from(cryptoSetting.key);
-    const cipher = crypto.createCipheriv(algo, key, iv);
+    const cipher: CipherGCM = crypto.createCipheriv(algo, key, iv) as CipherGCM;
```

Comment:
The type definition and the cast is a trick to make TS recognise the authentication related methods.

Reply:
> The type definition and the cast is a trick to make TS recognise the authentication related methods.

Should it be removed as we don't rely to a specific algo?

Reply:
> > The type definition and the cast is a trick to make TS recognise the authentication related methods.
> 
> Should it be removed as we don't rely to a specific algo?

It can't be removed, otherwise the transpilation fails: CipherGCM type have the `(get|set)AuthTag()` method, Cipher don't. 
Enforcing that mode type (or CipherCCM, CipherOFB, ...) with authentication just make the transpilation works. The code already handle all operation modes with embedded authentication.

Reply:
Okay, thanks!